programs/candidate_staking/src/lib.rs

```diff
@@ -16,6 +26,31 @@ pub mod candidate_staking {
 
         Ok(())
     }
+
+    pub fn stake(ctx: Context<Stake>, job_ad_id: String, base_bump: u8, general_bump: u8, application_bump: u8, job_bump: u8, amount: u32) -> Result<()> {
+
+        let general_parameter = &mut ctx.accounts.general_account;
+        let job_parameter = &mut ctx.accounts.job_account;
+        let application_parameter= &mut ctx.accounts.application_account;
+
+        if general_parameter.mint == ctx.accounts.token_mint.key() {
```

Comment:
you can do this check inside the account macro already, right?

Reply:
yes it is possible

Reply:
can we change the name of `mint` to `mint authority` or something like this? Just to be 100% clear what `mint` is referring to here exactly? Or otherwise put a reference to the docs in the SPL program where `mint` is used to enhance the understanding of whoever is reading the code.

Reply:
token_mint is the mint address, the authority is mint authority

Reply:
sure i will add all the documentation once we are able to stake and unstake